node_modules/oae-principals/lib/api.user.js

```diff
@@ -723,3 +961,109 @@ var _setAdmin = function(ctx, adminType, isAdmin, principalId, callback) {
+        }
+
+        // Send an email to the specified e-mail address
+        var data = {
+            'tenant': ctx.tenant(),
+            'user': user,
+            'baseUrl': TenantsUtil.getBaseUrl(ctx.tenant()),
+            'skinVariables': require('oae-ui').getTenantSkinVariables(ctx.tenant().alias),
+            'token': token
+        };
+        EmailAPI.sendEmail('oae-principals', 'verify', user, data, {'hash': token, 'ignoreVerification': true}, callback);
+    });
+};
+
+/**
+ * Resend an email token
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {String}     userId          The id of the user for who to resend the email token
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+var resendEmailToken = module.exports.resendEmailToken = function(ctx, userId, callback) {
+    var validator = new Validator();
+    validator.check(userId, {'code': 400, 'msg': 'A valid user id must be provided'}).isUserId();
+    validator.check(null, {'code': 401, 'msg': 'You have to be logged in to be able to resend an email token'}).isLoggedInUser(ctx);
+    if (validator.hasErrors()) {
+        return callback(validator.getFirstError());
+    }
+
+    // Ensure that you need to either be the user for which a token is being sent or a tenant admin
+    var principalResource = AuthzUtil.getResourceFromId(userId);
+    if (ctx.user().id !== userId && !ctx.user().isAdmin(principalResource.tenantAlias)) {
+        return callback({'code': 401, 'msg': 'You are not authorized to resend an email token'});
+    }
+
+    // Get the user object
+    PrincipalsDAO.getPrincipal(userId, function(err, user) {
+        if (err) {
+            return callback(err);
+        }
+
+        // Send the email token
+        return sendEmailToken(ctx, user, callback);
+    });
+};
+
+/**
+ * Verify an email token
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {String}     token           The email token to verify
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+var verifyEmailToken = module.exports.verifyEmailToken = function(ctx, token, callback) {
+    if (!token) {
+        return callback({'code': 400, 'msg': 'Missing token'});
+    }
+
+    PrincipalsDAO.getUserViaEmailToken(token, function(err, user) {
+        if (err) {
+            return callback(err);
+        }
+
+        PrincipalsDAO.validateToken(token, user, callback);
+    });
+};
+
+/**
+ * Generate a simple email token
+ *
+ * @return {String}     A string containing random alpha-numerical characters
+ * @api private
+ */
+var _generateEmailToken = function() {
+    var alphabet = 'abcdefghiklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
```

Comment:
I'm assuming we don't want to use ShortId for this? If not, would this function be useful elsewhere if we abstracted it out?